packages/stream-model-instance-handler/src/model-instance-document-handler.ts

```diff
@@ -163,11 +163,18 @@ export class ModelInstanceDocumentHandler implements StreamHandler<ModelInstance
     )
 
     if (payload.header) {
-      throw new Error(
-        `Updating metadata for ModelInstanceDocument Streams is not allowed.  Tried to change metadata for Stream ${streamId} from ${JSON.stringify(
-          state.metadata
-        )} to ${JSON.stringify(payload.header)}\``
-      )
+      const { shouldIndex, ...others } = payload.header
+      const otherKeys = Object.keys(others)
+      if (otherKeys.length) {
+        throw new Error(
+          `Unsupported metadata changes for ModelInstanceDocument Stream ${streamId}: ${otherKeys.join(
+            ','
+          )}. Only the index argument can be changed.`
+        )
+      }
+      if (shouldIndex !== null) {
+        state.metadata.shouldIndex = shouldIndex ?? false
+      }
```

Comment:
Is it intentional to check for an explicit `null` here please?
I think the logic should be more about checking for `null` or `undefined`, no?
```suggestion
      if (shouldIndex != null) {
        state.metadata.shouldIndex = shouldIndex
      }
```

Reply:
First wrote about intentionality. Then realised I introduced even more complexity.